src/js/helpers/ProjectOverwriteHelpers.js

```diff
@@ -135,7 +215,113 @@ export const createExternalVerseEdit = (projectPath, verseBefore, verseAfter, bo
-  fs.outputJsonSync(path.join(verseEditsPath, newFilename.replace(/[:"]/g, '_')), verseEdit);
+  writeCheckData(verseEdit, verseEditsPath);
 };
+
+/**
+* @description gets the group data for the current verse from groupsDataReducer
+* @param {Object} groupsData
+* @param {Object} contextId
+* @return {object} group data object.
+*/
+export function getGroupDataForVerse(groupsData, contextId) {
+  const filteredGroupData = {};
+  for (let groupItemKey of Object.keys(groupsData)) {
+    const groupItem = groupsData[groupItemKey];
+    if (groupItem) {
+      for (let check of groupItem) {
+        try {
+          if (isEqual(check.contextId.reference, contextId.reference)) {
+            if (!filteredGroupData[groupItemKey]) {
+              filteredGroupData[groupItemKey] = [];
+            }
+            filteredGroupData[groupItemKey].push(check);
+          }
+        } catch (e) {
+          console.warn(`Corrupt check found in group "${groupItemKey}"`, check);
+        }
+      }
+    }
+  }
+  return filteredGroupData;
+}
+
+/**
+ * returns true if contextIds are a match for reference and group
+ * @param {Object} contextId1
+ * @param {Object} contextId2
+ * @return {boolean}
+ */
+export function sameContext(contextId1, contextId2) {
+  if (!!contextId1 && !!contextId2) {
+    return isEqual(contextId1.reference, contextId2.reference) &&
+      (contextId1.groupId === contextId2.groupId);
+  }
+  return false;
+}
+
+export function getSelectionsFromChapterAndVerseCombo(bookId, chapter, verse, projectSaveLocation, quote = "") {
+  let selectionsObject = {};
+  const contextId = {
+    reference: {
+      bookId,
+      chapter,
+      verse
+    }
+  };
+  const selectionsPath = generateLoadPath(projectSaveLocation, contextId, 'selections');
+
+  if (fs.existsSync(selectionsPath)) {
+    let files = fs.readdirSync(selectionsPath);
+    files = files.filter(file => { // filter the filenames to only use .json
+      return path.extname(file) === '.json';
+    });
+    let sorted = files.sort().reverse(); // sort the files to use latest
+    if (quote) {
+      sorted = sorted.filter((filename) => {
+        const currentSelectionsObject = fs.readJsonSync(path.join(selectionsPath, filename));
+        return JSON.stringify(currentSelectionsObject.contextId.quote) === JSON.stringify(quote);
+      });
+    }
+    const filename = sorted[0];
+    selectionsObject = fs.readJsonSync(path.join(selectionsPath, filename));
+  }
+  return selectionsObject;
+}
+
+
+/**
+ * Generates the output directory.
+ * @param {Object} state - store state object.
+ * @param {String} checkDataName - checkDate folder name where data will be saved.
+ *  @example 'comments', 'reminders', 'selections', 'verseEdits' etc.
+ * @return {String} save path
+ */
+export function generateLoadPath(projectSaveLocation, contextId, checkDataName) {
```

Comment:
Isnt this function a duplicate ?

Reply:
The function I took this from is inside the `checking-tool-wrapper`

Reply:
Check CheckDataLoadActions.js

Reply:
Line 17

Reply:
I see, I will update